crates/core_arch/build.rs

```diff
@@ -1,3 +1,372 @@
+                    let function = format!(
+                        r#"
+{}
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[cfg_attr(all(test, target_arch = "arm"), assert_instr({}))]
+#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr({}))]
+pub unsafe fn {}(a: {}, b: {}) -> {} {{
+    {}(a, b)
+}}
+"#,
+                        current_comment,
+                        current_arm,
+                        current_aarch64,
+                        name,
+                        in_t,
+                        in_t,
+                        out_t,
+                        current_fn,
+                    );
+                    let test = format!(
+                        r#"
+    #[simd_test(enable = "neon")]
+    unsafe fn test_{}() {{
+        let a{};
+        let b{};
+        let e{};
+        let r: {} = transmute({}(transmute(a), transmute(b)));
+        assert_eq!(r, e);
+    }}
+            "#,
+                        name,
+                        values(globla_t, &a),
+                        values(globla_t, &b),
+                        values(globla_ret_t, &e),
+                        globla_ret_t,
+                        name
+                    );
+                    out_arm.push_str(&function);
+                    tests_arm.push_str(&test);
+                } else {
+                    let current_aarch64 = current_aarch64.clone().unwrap();
+                    let function = format!(
+                        r#"
+{}
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr({}))]
+pub unsafe fn {}(a: {}, b: {}) -> {} {{
+    {}(a, b)
+}}
+            "#,
+                        current_comment, current_aarch64, name, in_t, in_t, out_t, current_fn,
+                    );
+                    let test = format!(
+                        r#"
+    #[simd_test(enable = "neon")]
+    unsafe fn test_{}() {{
+        let a{};
+        let b{};
+        let e{};
+        let r: {} = transmute({}(transmute(a), transmute(b)));
+        assert_eq!(r, e);
+    }}
+            "#,
+                        name,
+                        values(globla_t, &a),
+                        values(globla_t, &b),
+                        values(globla_ret_t, &e),
+                        globla_ret_t,
+                        name
+                    );
+                    out_aarch64.push_str(&function);
+                    tests_aarch64.push_str(&test);
+                }
+            }
+        }
+    }
+    tests_arm.push('}');
+    tests_aarch64.push('}');
+
+    let meta_in = fs::metadata(IN)?;
+    let meta_arm = fs::metadata(ARM_OUT)?;
+    let meta_aarch64 = fs::metadata(AARCH64_OUT)?;
+    if meta_in.modified()? > meta_arm.modified()? {
+        let mut file_arm = File::create("src/arm/neon/generated.rs")?;
+        file_arm.write_all(out_arm.as_bytes())?;
```

Comment:
This always writes these files when building, but these files should only be written to when explicitly requested. Also when explicitly requested, the build should verify that the files that exist are the same as those that we already have (e.g. to detect on CI when somebody modified a file but forgot to commit updates to the generated ones).

Reply:
It doesn't always writes them, it writes them only when the spec file is newer then the generated file. Same as a make target.

Reply:
Will this run when a user uses this crates from crates.io as a dependency? That is, after cargo has downloaded the crate sources ?

Reply:
Will this incur a compilation cost on crates.io users?

Reply:
Moved this to an example so it needs to be manually ran & committed so it's only affecting people purposefully adding intrinsics.